proxy/proxy.go

```diff
@@ -25,6 +29,28 @@ var (
 	listenAddress = kingpin.Flag("web.listen-address", "Address to listen on for proxy and client requests.").Default(":8080").String()
 )
 
+var (
+	httpAPICounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "pushproxy_http_requests_total",
+			Help: "Number of http api requests.",
+		},
+		[]string{"code", "method", "path"},
+	)
+
+	httpProxyCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "pushproxy_proxied_requests_total",
+			Help: "Number of http proxy requests.",
+		},
+		[]string{"code", "method"},
+	)
+
+	httpCounterPush    = httpAPICounter.MustCurryWith(prometheus.Labels{"path": "/push"})
```

Comment:
This would be better handled once in the mux/handler, rather than duplicating this for every handler.